sufia-models/app/models/concerns/sufia/collection_behavior.rb

```diff
@@ -15,10 +15,29 @@ def update_permissions
       self.visibility = "open"
     end
 
-    # Compute the sum of each file in the collection
+    # Compute the sum of each file in the collection using Solr to
+    # avoid having to proces GBs of data for a simple number. Can be
+    # cleaned up once it is confirmed to work
+    #
     # Return an integer of the result
     def bytes
-      members.reduce(0) { |sum, gf| sum + gf.content.size.to_i }
+      query = "has_model_ssim:GenericFile"
+      # Cribbed from Hydra::Collections to speed up the resolution
+      # process
+      args = {
+        :fq => "{!join from=hasCollectionMember_ssim to=id}id:#{self.id}",
+        :fl => "id,file_size_isi",
+        :rows => self.members.count
+      }
+
+      files = ActiveFedora::SolrService.query(query, args)
+
+      byte_count = 0
```

Comment:
@narogers These 5 lines could become  files.reduce(0) { |sum, f| sum + f['file_size_isi'] }   Then you would not need the local variable.